src/Corax/Querying/Matches/SortingMatches/SortingMultiMatch.Helpers.cs

```diff
@@ -106,6 +113,20 @@ private struct EntryComparerHelper
             else
                 indexes.Sort(new IndirectComparer<Descending<TComparer1>, TComparer2, TComparer3>(ref match, batchTerms, new(comparer1), comparer2, comparer3, true));
 
+            if (typeof(TComparer1) != typeof(EntryComparerByScore) || typeof(TComparer1) != typeof(Descending<EntryComparerByScore>))
+            {
+                if (match._sortingDataTransfer.IncludeScores && match._scoreBufferHandler != null)
+                {
+                    match._scoresResults.EnsureCapacityFor(indexes.Length);
+                    foreach (var t in indexes)
+                        match._scoresResults.AddByRefUnsafe() = match._secondaryScoreBuffer[t];
+
+                    match._secondaryScoreBuffer = default;
+                    match._scoreBufferHandler.Dispose();
+                    match._scoreBufferHandler = null;
+                }
+                
+            }
         }
```

Comment:
This file is a bit unapproachable if you don't know exactly what you are doing. Some comments over the what type of sorting operation each one is handling would be very useful for "Future us".

Reply:
@maciejaszyk can you please create a dedicated issue for that so we'll handle it in separate PR